src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoNullable.cs

```diff
@@ -61,16 +62,54 @@ internal override void Read(JsonTokenType tokenType, JsonSerializerOptions optio
 
         internal override void ReadEnumerable(JsonTokenType tokenType, JsonSerializerOptions options, ref ReadStack state, ref Utf8JsonReader reader)
         {
-            if (ValueConverter != null)
+            if (ValueConverter == null || !ValueConverter.TryRead(typeof(TProperty), ref reader, out TProperty value))
+            {
+                ThrowHelper.ThrowJsonReaderException_DeserializeUnableToConvertValue(RuntimePropertyType, reader, state);
+                return;
+            }
+
+            // Converting to TProperty? here lets us share a common ApplyValue() with ApplyNullValue().
+            TProperty? nullableValue = new TProperty?(value);
+            ApplyValue(nullableValue, options, ref state.Current);
+        }
+
+        // If this method is changed, also change JsonPropertyInfoNotNullable.ApplyValue and JsonSerializer.ApplyObjectToEnumerable
+        private void ApplyValue(TProperty? value, JsonSerializerOptions options, ref ReadStackFrame frame)
```

Comment:
Could this be a static helper method instead that is shared? At quick glance, it only cares about the arguments being passed in.

Reply:
Done. Combined the two property methods into one. There is a tiny perf hit for a `null` value case.